File: cpp/raindrops/raindrops.cpp
#include "raindrops.h"
#include <map>

std::map<int, std::string> rain {{3, "Pling"}, {5, "Plang"}, {7, "Plong"}};

std::string raindrops::convert(int n) {
  std::string res;
  std::map<int, std::string>::iterator it;

  for (it = rain.begin(); it != rain.end(); it++) {
    int key = it->first;
    if (n % key == 0) {
      res += rain[key];
    }
  }
  return res.length() == 0 ? std::to_string(n) : res;
}

File: typescript/raindrops/raindrops.ts
const translator: [number, string][] = [
  [3, 'Pling'],
  [5, 'Plang'],
  [7, 'Plong'],
]

class Raindrops {
  convert(n: number) {
    let res = translator.reduce((accum: string, entry: [number, string]) => {
      if (n % entry[0] === 0) {
        accum += entry[1]
      }
      return accum
    }, '')
    return res === '' ? n.toString() : res
  }
}

export default Raindrops